polkadot/xcm/src/v2/mod.rs

```diff
@@ -262,6 +267,8 @@ pub const VERSION: super::Version = 2;
 /// An identifier for a query.
 pub type QueryId = u64;
 
+/// DEPRECATED. Please use XCMv3 or XCMv4 instead.
+#[deprecated(note = "XCMv2 will be removed after June 2024. Please use XCMv3 or XCMv4 instead.")]
```

Comment:
when v5 won't be ready in the June, will we still remove v2? :)
I would say, still, we remove v2 when adding v5 in the same branch.

Reply:
The notice is saying it won't be removed before that date, but it can certainly be kept for longer after that date, so we're covered.
It would be good to have it ready around june though 😅